File: libs/document-processor/src/repo/core/schema/JavaScriptCode.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId } from '@blue-labs/language';

export const JavaScriptCodeSchema = withTypeBlueId(blueIds['JavaScript Code'])(
  z.object({
    code: z.string().optional(),
  })
);

export type JavaScriptCode = z.infer<typeof JavaScriptCodeSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/JavaScriptCode.java
package blue.language.types.conversation;

import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;

@TypeAlias("Conversation/JavaScript Code")
@TypeBlueId("3hYcmWMtMUPAzXBLFLb7BpuG9537tuTJPCr7pxWXvTZK")
public class JavaScriptCode {
    public String code;

    public JavaScriptCode code(String code) {
        this.code = code;
        return this;
    }
}
